core/src/base/mod.rs

```diff
@@ -1,2 +1,22 @@
 mod link;
 pub use link::{Link, OpenGraph};
+use serde::{Deserialize, Serialize};
+/// ARK Config
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Config {
+    /// Sorting mode.
+    pub mode: Mode,
+    /// The score content, repensented to a score.
+    pub score: Option<Vec<String>>,
```

Comment:
This is wrong:
1) `scores` must be separate from config, it is user-defined data which is persisted in the folder with bookmarks. See https://github.com/ARK-Builders/ARK-Shelf-Desktop/issues/9 for details:
> Proper implementation requires https://github.com/ARK-Builders/arklib/issues/16. While it isn't there yet, .ark/shelf/scores file can be used for storing scores instead of .ark/scores.

`$FOLDER/.ark/shelf/scores` path must be used to store scores.

At the same time, `mode` is user-defined style of data representation, it must be in device-specific location, e.g. `$HOME/.ark` or whatever (not really important, we also don't really need to store it, we can just have default mode).

2) `scores` must be mapping from `ResourceId` to `Integer`. Type `Vec<String>` looks missing the point.